packages/aws-cdk-lib/aws-secretsmanager/lib/rotation-schedule.ts

```diff
@@ -125,7 +125,10 @@ export class RotationSchedule extends Resource {
     }
 
     let automaticallyAfterDays: number | undefined = undefined;
-    if (props.automaticallyAfter?.toMilliseconds() !== 0) {
+    if (props.automaticallyAfter && props.automaticallyAfter.toDays() > 1000) {
+      throw new Error(`automaticallyAfter must not be greater than 1000 days, got ${props.automaticallyAfter.toDays()} days`);
+    }
+    if (!props.automaticallyAfter || props.automaticallyAfter.toMilliseconds() !== 0) {
```

Comment:
```suggestion
    if (props.automaticallyAfter?.toMilliseconds() !== 0) {
```
Since they are equivalent I think it's better to keep the concise version (unless I'm missing something).

Reply:
https://github.com/aws/aws-cdk/pull/27570/commits/eb2392dcb97be0ebbd22f30ed5bf056dad112e24

Reply:
https://github.com/aws/aws-cdk/pull/27592/files